### PR TITLE
JN-741 Deselect participants when successfully requesting sample kits

### PR DESCRIPTION
**Pull request by kentmw**
#### DESCRIPTION
If any kits have been successfully requested for participants, it will now deselect all participants from the KitEnrolleeSelection view. If none were successful, the current selection will be maintained

#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*
Go to https://localhost:3000/ourhealth/studies/ourheart/env/sandbox/kits/eligible
and select one or more participants and click "Send sample collection kit". In the modal that appears, press "Request Kits". If a kit was sent successfully, the table should have no one selected. If no kits were successful, it should keep the selection

### CODE REVIEW

File: ui-admin/src/study/kits/RequestKitsModal.tsx [hunk 1]
```diff
@@ -37,8 +38,9 @@ export default function RequestKitsModal({
         Store.addNotification(successNotification(
                     `${response.kitRequests.length} kit requests created`
         ))
+        success = true
       }
-      onSubmit()
+      onSubmit(success)
```
Comment (devonbush) [inline ui-admin/src/study/kits/RequestKitsModal.tsx]: I don't think you need the intermediate variable
```suggestion
      onSubmit(!!response.kitRequests.length)
```

File: ui-admin/src/study/kits/KitEnrolleeSelection.tsx [hunk 2]
```diff
@@ -69,9 +69,12 @@ export default function KitEnrolleeSelection({ studyEnvContext }: { studyEnvCont
     setEnrollees(enrolleeRows)
   }, [studyEnvContext.study.shortcode, studyEnvContext.currentEnv.environmentName])
```
Comment (devonbush) [inline ui-admin/src/study/kits/KitEnrolleeSelection.tsx:71]:
```suggestion
/** indicate success as true if any kits were successfully created */
```

File: ui-admin/src/study/kits/KitEnrolleeSelection.tsx [hunk 3]
```diff
@@ -69,9 +69,12 @@ export default function KitEnrolleeSelection({ studyEnvContext }: { studyEnvCont
...
 
-  const onSubmit = async () => {
+  const onSubmit = async (success: boolean) => {
     setShowRequestKitModal(false)
     reload()
+    if (success) {
```
Comment (devonbush) [inline ui-admin/src/study/kits/KitEnrolleeSelection.tsx]:
```suggestion
   /**  if any kits were created, that changes the filter state of the table
       and could result in hidden items still being selected.  Clear the selections to be safe */
```

**Review by devonbush — COMMENTED** [review]
nice -- this fixes a massive problem.  Couple of nits and comment suggestions

File: ui-admin/src/study/kits/KitEnrolleeSelection.tsx (hunk 2 above)
Comment (kentmw) [reply ui-admin/src/study/kits/KitEnrolleeSelection.tsx:71]: I changed the variable name to be more clear. Let me know if you want the comment anyway for further clarity.

File: ui-admin/src/study/kits/KitEnrolleeSelection.tsx (hunk 3 above)
Comment (kentmw) [reply ui-admin/src/study/kits/KitEnrolleeSelection.tsx]: Did you mean to suggest removing the if condition?

File: ui-admin/src/study/kits/RequestKitsModal.tsx (hunk 1 above)
Comment (kentmw) [reply ui-admin/src/study/kits/RequestKitsModal.tsx]: Done

File: ui-admin/src/study/kits/KitEnrolleeSelection.tsx (hunk 2 above)
Comment (devonbush) [reply ui-admin/src/study/kits/KitEnrolleeSelection.tsx:71]: nice, good change

**Review by devonbush — APPROVED** [review]

**Review by MatthewBemis — APPROVED** [review]

**Pull request merged**